packages/protocol-http/src/extensions/http.ts

```diff
@@ -0,0 +1,42 @@
+import { HttpHandler } from "../httpHandler";
+import { HttpHandlerExtensionConfiguration } from "./httpExtensionConfiguration";
+
+/**
+ * @internal
+ */
+export const getHttpHandlerConfiguration = <HandlerConfig extends object = Record<string, unknown>>(
+  runtimeConfig: Partial<{ httpHandler: HttpHandler<HandlerConfig> }>
+) => {
+  let httpHandler = runtimeConfig.httpHandler!;
+  const httpHandlerConfigs: HandlerConfig = httpHandler.httpHandlerConfigs();
+  return {
+    setHttpHandler(handler: HttpHandler<HandlerConfig>): void {
+      httpHandler = handler;
+    },
+    httpHandler(): HttpHandler<HandlerConfig> {
+      return httpHandler;
+    },
+    updateHttpClientConfig(key: keyof HandlerConfig, value: HandlerConfig[typeof key]): void {
+      httpHandlerConfigs[key] = value;
```

Comment:
should this be:
```
this.httpHandler().httpHandlerConfigs()[key] = value
```
?

This reference looks like it will become stale after a call to setHttpHandler